README.md

```diff
@@ -1,15 +1,170 @@
+
+In addition we will use a wireframe navigation model and a service locator example[[5] [di]].
 
 ## Usage
 
-**_TODO_**
+This framework contains all the classes needed to implement your presentation logic following the MVP pattern. To use the view package, make your ``ViewController`` extend from ``Botham ViewController`` and specify in the storyboard wich class and Storyboard ID is linked to:
+
+```swift
+import BothamUI
+
+class SampleViewController: BothamViewController {
+    /*...*/
+}
+```
+![storyboardReference]
+
+#### Storyboard
+
+`BothamStoryboard` provide a series of methods to help you instantiate view controllers by there storyboard ID. By default `instantiateViewController()` will search for view controller with the storyboard ID with the same name as the class.
+
+```swift
+import BothamUI
+
+let mainStoryboard = BothamStoryboard(name: "Main")
+let viewController: SampleViewController = mainStoryboard.instantiateViewController("SampleViewController")
+let viewController: SampleViewController = mainStoryboard.instantiateViewController()
+```
+
+#### Presenter
+To follow the MVP pattern, BothamUI also provides a ``BothamPresenter`` protocol that will be responsible for all your presentation logic. BothamUI will take care of linking your view (a ``BothamViewController``) with your presenter and subscribing it to its lifecycle. In order to do that, create a class that implement ``BothamPresenter`` and link it to your view:
+
+```swift
+import BothamUI
+
+class SamplePresenter: BothamPresenter {
+    private weak var ui: SampleUI?
+
+    init(ui: CharacterDetailUI) {
+        self.ui = ui
+    }
+    
+    func viewDidLoad() {
+        /* ... */
+    }
+}
+```
+
+```swift
+protocol SampleUI: BothamUI {
+	/* ... */
+}
+```
+```swift
+class SampleViewController: BothamViewController, SampleUI {
+    /*...*/
+}
+```
+
+#### Service Locator
+
+We recommend the usage of a Service Locator pattern in order to instantiate view controllers.
+
+```swift
+class ServiceLocator {
+
+    static let sharedInstance = ServiceLocator()
+
+    func provideCharacterDetailViewController() -> CharacterDetailViewController {
+        let viewController: CharacterDetailViewController = provideMainStoryboard().viewController()
+        viewController.presenter = CharacterDetailPresenter(ui: viewController)
+        return viewController
+    }
+}
+```
+
+#### Lifecycle
+
+Once both, view and presenter, are linked you can react to your view lifecycle directly from the presenter. You will be also able to call your view easily from the presenter:
+
+```swift
+class SamplePresenter: BothamPresenter {
+    private weak var ui: SampleUI?
+
+    init(ui: CharacterDetailUI) {
+        self.ui = ui
+    }
+    
+    func viewDidLoad() {
+        self.ui?.showMessage("Welcome to Botham")
+    }
+}
+```
+
+To understand when the lifecycle methods are called take a look at the following table:
 
```

Comment:
Are we going to talk about data sources? If I'm not wrong we have a BothamDataSource.